infra/util/Crc.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef INFRA_CRC_HPP
+#define INFRA_CRC_HPP
+
+#include "infra/util/ByteRange.hpp"
+#include <climits>
+#include <cstddef>
+#include <cstdint>
+
+namespace infra
+{
+    template<typename CRC_TYPE>
+    constexpr CRC_TYPE PolyReverse(CRC_TYPE Polynomial);
+
+    template<typename CRC_TYPE, CRC_TYPE Polynomial, CRC_TYPE InitValue = 0, CRC_TYPE FinalXor = 0, bool ReflectInput = false, bool ReflectOutput = false>
+    class Crc
+    {
+        static_assert(ReflectInput == ReflectOutput, "Currently, non-matching ReflectInput and ReflectOutput is not supported");
+
+    public:
+        Crc() = default;
+        ~Crc() = default;
```

Comment:
These `default`ed constructor and destructor can be omitted

Reply:
removed